api/src/test/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/LayerTest.kt

```diff
@@ -0,0 +1,90 @@
+package org.jetbrains.kotlinx.dl.api.core.layer
+
+import org.jetbrains.kotlinx.dl.api.core.KGraph
+import org.jetbrains.kotlinx.dl.api.core.shape.toLongArray
+import org.junit.jupiter.api.Assertions
+import org.tensorflow.*
+import org.tensorflow.op.Ops
+import java.lang.IllegalArgumentException
+
+open class LayerTest {
+
+    protected fun createFloatArray1D(shape: LongArray): FloatArray {
+        return FloatArray(shape[0].toInt())
+    }
+
+    protected fun createFloatArray2D(shape: LongArray): Array<FloatArray> {
+        return Array(shape[0].toInt()) {
+            createFloatArray1D(shape.drop(1).toLongArray())
+        }
+    }
+
+    protected fun createFloatArray3D(shape: LongArray): Array<Array<FloatArray>> {
+        return Array(shape[0].toInt()) {
+            createFloatArray2D(shape.drop(1).toLongArray())
+        }
+    }
+
+    protected fun createFloatArray4D(shape: LongArray): Array<Array<Array<FloatArray>>> {
+        return Array(shape[0].toInt()) {
+            createFloatArray3D(shape.drop(1).toLongArray())
+        }
+    }
+
+    protected fun createFloatArray5D(shape: LongArray): Array<Array<Array<Array<FloatArray>>>> {
+        return Array(shape[0].toInt()) {
+            createFloatArray4D(shape.drop(1).toLongArray())
+        }
+    }
+
+    private fun getInputOp(tf: Ops, input: Array<*>, inputDim: Int): Operand<Float> {
```

Comment:
And this @mkaze can be achieved I think with the helper functions present in ShapeFunctions which are used for casting arrays. Are they suitable for these tests?